test/Microsoft.Azure.WebJobs.Host.EndToEndTests/InvocationFilterEndToEndTests.cs

```diff
@@ -0,0 +1,635 @@
+        }
+    }
+
+    public static class ClassWithFunctionToTest
+    {
+        [NoAutomaticTrigger]
+        public static void AnotherFunction(FunctionExecutingContext executingContext)
+        {
+            executingContext.Logger.LogInformation("AnotherFunction invoked!");
+        }
+    }
+
+    public class TestClassInterface : IFunctionInvocationFilter
+    {
+        [NoAutomaticTrigger]
+        [InvokeFunctionFilter(executingFilter: "TestClassInterface")]
+        public void TestInterface(string input, ILogger logger)
+        {
+            logger.LogInformation("TestInterface invoked!");
+        }
+
+        public Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            executingContext.Logger.LogInformation("Class.OnExecutingAsync invoked!");
+            return Task.CompletedTask;
+        }
+
+        public Task OnExecutedAsync(FunctionExecutedContext executedContext, CancellationToken cancellationToken)
+        {
+            executedContext.Logger.LogInformation("Class.OnExecutedAsync invoked!");
+            return Task.CompletedTask;
+        }
+    }
+
+    public class TestLoggingFilter : InvocationFilterAttribute
+    {
+        public override Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            executingContext.Logger.LogInformation("Test executing!");
+            return Task.CompletedTask;
+        }
+
+        public override Task OnExecutedAsync(FunctionExecutedContext executedContext, CancellationToken cancellationToken)
+        {
+            executedContext.Logger.LogInformation("Test executed!");
+            return Task.CompletedTask;
+        }
+    }
+
+    public class TestUserAuthorizationFilter : InvocationFilterAttribute
+    {
+        public string AllowedUsers { get; set; }
+
+        public override Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            executingContext.Logger.LogInformation("Test executing!");
+
+            if (!AllowedUsers.Contains("Admin"))
+            {
+                executingContext.Logger.LogInformation("This is an unauthorized user!");
+                throw new Exception("Not Allowing Unauthorized Users!");
+            }
+
+            executingContext.Logger.LogInformation("This is an authorized user!");
+            return Task.CompletedTask;
+        }
+    }
+
+    public class HTTPRequestFilter : InvocationFilterAttribute
+    {
+        public HttpRequestMessage HttpRequestToValidate { get; set; }
+
+        public override Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            executingContext.Logger.LogInformation("Test executing!");
+
+            IReadOnlyDictionary<string, object> arguments = executingContext.Arguments;
```

Comment:
Fix this code. E.g. try/cast the first argument value to an HttpRequestMessage, then you can party on the query params/headers as needed